security/advancedtls/advancedtls.go

```diff
@@ -475,3 +479,27 @@ func cloneTLSConfig(cfg *tls.Config) *tls.Config {
 	}
 	return cfg.Clone()
 }
+
+// GetCertificateWithSNI returns the certificate that matches the SNI field
+// based on the given ClientHelloInfo if GetCertificate returns a list of
+// certificates
+func (o *ServerOptions) GetCertificateWithSNI(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
```

Comment:
This will become a public API. Can we make it internal?

Reply:
Does it mean I need to change `GetCertificateWithSNI` to `getCertificateWithSNI`?

Reply:
yes.